Mathlib/FieldTheory/PerfectClosure.lean

```diff
@@ -8,7 +8,41 @@ import Mathlib.FieldTheory.Perfect
 #align_import field_theory.perfect_closure from "leanprover-community/mathlib"@"70fd9563a21e7b963887c9360bd29b2393e6225a"
 
 /-!
-# The perfect closure of a field
+
+# The perfect closure of a characteristic `p` ring
+
+## Main definitions
+
+- `PerfectClosure`: the perfect closure of a characteristic `p` ring, which is the smallest
+  extension that makes frobenius surjective.
+
+- `PerfectClosure.mk K p (n, x)` for `n : ℕ` and `x : K` is an element of `PerfectClosure K p`,
+  viewed as `x ^ (p ^ -n)`. Every element of `PerfectClosure K p` is of this form
+  (`PerfectClosure.mk_surjective`).
+
+- `PerfectClosure.of`: the structure map from `K` to `PerfectClosure K p`.
+
+- `PerfectClosure.lift`: given a ring `K` of characteristic `p` and a perfect ring `L` of the same
+  characteristic, any homomorphism `K →+* L` can be lifted to `PerfectClosure K p`.
+
+## Main results
+
+- `PerfectClosure.induction_on`: to prove a result for all elements of the prefect closure, only
+  need to prove it for all `x ^ (p ^ -n)`.
+
+- `PerfectClosure.mk_mul_mk`, `PerfectClosure.one_def`, `PerfectClosure.mk_add_mk`,
+  `PerfectClosure.neg_mk`, `PerfectClosure.zero_def`, `PerfectClosure.mk_zero_zero`,
+  `PerfectClosure.mk_zero`, `PerfectClosure.mk_inv`, `PerfectClosure.mk_pow`:
+  how to do multiplication, addition, etc. on elements of form `x ^ (p ^ -n)`.
+
+- `PerfectClosure.eq_iff'`: when does `x ^ (p ^ -n)` equal.
```

Comment:
I think we can also ignore it, but the name should be `PerfectClosure.mk_eq_iff`.

Reply:
In fact the existing API confuses me, there is also `PerfectClosure.eq_iff` but which assumes moreover that `K` is reduced (and whose statement uses `Quot.mk` but not `PerfectClosure.mk` --- aren't they defeq ??? very strange), so the criterion becomes simpler, while the `PerfectClosure.eq_iff'` is stated for all rings.